gprofiler/profilers/perf.py

```diff
@@ -354,7 +354,8 @@ def _generate_profile_data(self, stacks: StackToSampleCount, pid: int) -> Profil
             appid = self._get_appid(pid)
         else:
             appid = None
-        return ProfileData(stacks, appid, metadata)
+        container_name = self._profiler_state.get_container_name(pid)
+        return ProfileData(stacks, appid, metadata, container_name)
```

Comment:
```suggestion
        return ProfileData(stacks, appid, metadata, self._profiler_state.get_container_name(pid))
```

No need for the extra variable, if it's only used once